sentry/src/routes/campaign.rs

```diff
@@ -20,25 +38,110 @@ pub async fn create_campaign<A: Adapter>(
     let error_response = ResponseError::BadRequest("err occurred; please try again later".into());
 
     // insert Campaign
-    
-    // match insert_campaign(&app.pool, &campaign).await {
-    //     Err(error) => {
-    //         error!(&app.logger, "{}", &error; "module" => "create_channel");
-
-    //         match error {
-    //             PoolError::Backend(error) if error.code() == Some(&SqlState::UNIQUE_VIOLATION) => {
-    //                 Err(ResponseError::Conflict(
-    //                     "channel already exists".to_string(),
-    //                 ))
-    //             }
-    //             _ => Err(error_response),
-    //         }
-    //     }
-    //     Ok(false) => Err(error_response),
-    //     _ => Ok(()),
-    // }?;
+
+    match insert_or_modify_campaign(&app.pool, &campaign, &app.redis).await {
+        Err(error) => {
+            // error!(&app.logger, "{}", &error; "module" => "create_channel");
+            return Err(ResponseError::Conflict("channel already exists".to_string()));
+        }
+        Ok(false) => Err(error_response),
+        _ => Ok(()),
+    }?;
 
     let create_response = SuccessResponse { success: true };
 
     Ok(success_response(serde_json::to_string(&campaign)?))
 }
+
+// TODO: Double check redis calls
+async fn get_spent_for_campaign(redis: &MultiplexedConnection, id: CampaignId) -> Result<UnifiedNum, PoolError> {
+    let key = format!("adexCampaign:campaignSpent:{}", id);
+    // campaignSpent tracks the portion of the budget which has already been spent
+    let campaign_spent = match redis::cmd("GET")
+    .arg(&key)
+    .query_async::<_, Option<String>>(&mut redis.clone())
+    .await?
+    {
+        Some(spent) => UnifiedNum::from(spent),
+        // TODO: Double check if this is true
+        // If the campaign is just being inserted, there would be no entry therefore no funds would be spent
```

Comment:
How can we obtain this from Postgres?

Reply:
I will double check, but probably we would need to use analytics, since the spendable & last approved state are based on the Channel and not the Campaign.

PS: We haven't changed Analytics according to AIP#61 yet.

Reply:
Skip this. If there is no record in Redis then something is gone terribly wrong.